WordPress/wordpress.org
# Add a translations' bulk update tool for the GTE in the Translation Consistency

One of the most requested functionalities by GTEs is the massive update of translations. We talked about this in this P2: [Bulk updates in the Translation Consistency](https://make.wordpress.org/polyglots/2023/05/15/bulk-updates-in-the-translation-consistency/).

This PR adds a new tool, only available for the GTE of each locale, to make bulk updates:

![image](https://github.com/WordPress/wordpress.org/assets/1667814/e46b2e3b-c95e-4f34-8efa-7cc23ee70d03)

If the logged user is not a GTE, she doesn't get the bulk update form:

![image](https://github.com/WordPress/wordpress.org/assets/1667814/2e7248f8-c38e-4a41-92d0-12e737499da1)

Once the GTE makes a bulk update, she gets a list with all the translations and a link to review each new translation.

![image](https://github.com/WordPress/wordpress.org/assets/1667814/7db5f230-693f-4526-9da9-43162a6727d9)

if you have updated all the strings with the same translation, you are going to see a message like this:

![image](https://github.com/WordPress/wordpress.org/assets/1667814/8c0b786e-d7fa-4c9c-896f-db0f0d9d60c4)

If all the translations are equal, the GTE will not see any form:

![image](https://github.com/WordPress/wordpress.org/assets/1667814/33b51b3a-27da-4473-ab5a-f3e6e906e125)

## Testing

I have tested this PR with the emoji language, to avoid putting fake translations in a real locale. If you are emoji GTE, you can use the next links to test the functionality (if you are GTE in another language, feel free to test it there).

To test it:

- Check the translations you want to update.
- Select the translation you want to use as replacement.
- Click the "Update translations" button.
- Follow the links with the updates.

- [All translations translating the "Continue" word](https://translate.wordpress.org/consistency/?search=continue&set=art-xemoji%2Fdefault&project=).
  - [Plugins - All-in-One WP Migration - Development (trunk)](https://translate.wordpress.org/projects/wp-plugins/all-in-one-wp-migration/dev/art-xemoji/default/?filters[status]=either&filters[original_id]=6519552&filters[translation_id]=111009434)
  - [Plugins - Starter Templates — Elementor, WordPress & Beaver Builder Templates - Development (trunk)](https://translate.wordpress.org/projects/wp-plugins/astra-sites/stable/art-xemoji/default/?filters[status]=either&filters[original_id]=12904513&filters[translation_id]=110521591)
  - [WordPress - 6.2.x - Administration](https://translate.wordpress.org/projects/wp/6.2.x/admin/art-xemoji/default/?filters[status]=either&filters[original_id]=16084715&filters[translation_id]=111011298)

- [All translations translating the "Delete %s" word](https://translate.wordpress.org/consistency/?search=Delete+%25s&set=art-xemoji%2Fdefault&project=)
  - [Plugins - Elementor Website Builder – Development (trunk)](https://translate.wordpress.org/projects/wp-plugins/elementor/dev/art-xemoji/default/?filters[status]=either&filters[original_id]=5892745&filters[translation_id]=111032589)
  - [WordPress - 6.2.x - Administration](https://translate.wordpress.org/projects/wp/6.2.x/admin/art-xemoji/default/?filters[status]=either&filters[original_id]=16083959&filters[translation_id]=111032592)
  - [WordPress - 6.2.x - Administration](https://translate.wordpress.org/projects/wp/6.2.x/admin/art-xemoji/default/?filters[status]=either&filters[original_id]=16085958&filters[translation_id]=111032591)
  - [Plugins - Gutenberg - Development (trunk)](https://translate.wordpress.org/projects/wp-plugins/gutenberg/dev/art-xemoji/default/?filters[status]=either&filters[original_id]=12774652&filters[translation_id]=111032598)

- [`<a href="%s">Please update WordPress</a>.`](https://translate.wordpress.org/consistency/?search=%3Ca+href%3D%22%25s%22%3EPlease+update+WordPress%3C%2Fa%3E.&set=art-xemoji%2Fdefault&project=)
  - [WordPress - 6.2.x](https://translate.wordpress.org/projects/wp/6.2.x/art-xemoji/default/?filters[status]=either&filters[original_id]=16078591&filters[translation_id]=111032640)
  - [Plugins - Plugin Dependencies - Development (trunk)](https://translate.wordpress.org/projects/wp-plugins/wp-plugin-dependencies/dev/art-xemoji/default/?filters[status]=either&filters[original_id]=15818962&filters[translation_id]=111032702)
  - [Plugins - Jetpack – WP Security, Backup, Speed, & Growth - Development (trunk)](https://translate.wordpress.org/projects/wp-plugins/jetpack/dev/art-xemoji/default/?filters[status]=either&filters[original_id]=7682985&filters[translation_id]=111032726)
  - [Plugins - AI ChatBot - Development (trunk)](https://translate.wordpress.org/projects/wp-plugins/chatbot/dev/art-xemoji/default/?filters[status]=either&filters[original_id]=15206249&filters[translation_id]=111032733)

## Errors

If the bulk update detects some error, it doesn't make the bulk update and it shows a message like this:

![image](https://github.com/WordPress/wordpress.org/assets/1667814/336d388c-07eb-475f-9c8f-9f1768b6b75f)
